library/src/main/java/okhttp3/mock/matchers/BodyMatcher.java

```diff
@@ -0,0 +1,23 @@
+package okhttp3.mock.matchers;
+
+import okhttp3.Request;
+import okhttp3.mock.RequestCache;
+import java.util.regex.Pattern;
+
+
+public class BodyMatcher extends PatternMatcher  {
+
+    public BodyMatcher(Pattern pattern) {
```

Comment:
I'd use a more generic apprach like:
```
boolean matches(body: RequestBody)
```
 and `Pattern` can be just an implementation of it.

Reply:
yep this would make it more generic, would this be offered as a predicate of string as well?  that way , user of bodyMatcher would not need to write repeat the code to read the request body as a string (i.e. body.writeTo(buffer)